.github/workflows/protobuf.yml

```diff
@@ -0,0 +1,27 @@
+name: Protobuf Compatibility
+
+on:
+  pull_request:
```

Comment:
I'm wondering if we should do this on PRs. I'm guessing the runtime isn't very long? But given the low frequency with which we make edits to this area, I would think just having it in the merge queue would be plenty sufficient.